pkg/api/customization/catalog/validator.go

```diff
@@ -0,0 +1,42 @@
+package catalog
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+)
+
+var (
+	controlChars   = regexp.MustCompile("[[:cntrl:]]")
+	controlEncoded = regexp.MustCompile("%[0-1][0-9,a-f,A-F]")
+)
+
+func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	if url, ok := data["url"].(string); ok {
+		sanitizedURL, err := validateURL(url)
+		if err != nil {
+			return err
```

Comment:
Errors coming out of validators should be wrapped, use `httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("%v", err))`

Reply:
fixed